tvtk/wrapper_gen.py

```diff
@@ -1336,3 +1470,251 @@ def %(setter)s(self, arg):
+            t_def = 'traits.Trait(%(default)s, %(default)s, '\
+                    'traits.Range%(rng)s'%locals()
+            t_def = t_def[:-1] + ', enter_set=True, auto_set=False))'
+        else:
+            t_def = 'traits.Trait(%(default)s, '\
+                    'traits.Range%(rng)s'%locals()
+            t_def = t_def[:-1] + ', enter_set=True, auto_set=False))'
+
+        name = self._reform_name(vtk_attr_name)
+        vtk_set_meth = getattr(klass, 'Set' + vtk_attr_name)
+        self._write_trait(out, name, t_def, vtk_set_meth,
+                          mapped=False)
+
+
+    # ------------------------------------------------------
+    # Traits that need special handling
+    # ------------------------------------------------------
+    # To add a trait for special handling,
+    # add an item to the `special_traits` below,
+    # then add a method that handles the trait.
+    # The method name should be referred to by the `special_traits`
+    # mapping.
+
+    # special_traits = {
+    #        expr: (updateable, allow_to_fail, name_of_method)
+    #      }
+    # where
+    # `expr` is a regular expression pattern for matching
+    #         {vtkObject}.{Attribute}
+    # `updateable` is whether the trait should
+    #         be in the `updateable_traits` tuple,
+    # `allow_to_fail` is whether the trait should be in the
+    #         the `allow_update_failure` tuple
+    # `name_of_method` is the name of the method for writing
+    #         the code for this trait,
+    #         i.e. getattr(self, name_of_method)(...)
+    special_traits = {
+        '[a-zA-Z0-9]+\.Output$': (
+            False, False, '_write_any_output'),
+        '[a-zA-Z0-9]+\.Source$': (
+            False, False, '_write_any_source'),
+        '[a-zA-Z0-9]+\.ScalarType$': (
+            False, False, '_write_any_scalar_type'),
+
+        # In VTK > 4.5, Set/GetInput have multiple signatures
+        '[a-zA-Z0-9]+\.Input$': (
+            False, False, '_write_any_input'),
+
+        '[a-zA-Z0-9]+\.InputConnection$': (
+            False, False, '_write_any_input_connection'),
+        '[a-zA-Z0-9\.]+FileName$': (
+            True, False, '_write_any_something_file_name'),
+        '[a-zA-Z0-9\.]+FilePrefix$': (
+            True, False, '_write_any_something_file_prefix'),
+        'vtkImageReader2.HeaderSize$': (
+            True, False, '_write_image_reader2_header_size'),
+
+        # PropColorValue is not initialised, GetPropColorValue
+        # gives random values as a tuple of float[3] that are
+        'vtkHardwareSelector.PropColorValue$': (
+            True, True, '_write_hardware_selector_prop_color_value'),
+
+        # In VTK 5.8, tolerance is initialised as 0 while the range
+        # is 1-100
+        'vtkAxesTransformRepresentation.Tolerance$': (
+            True, True, '_write_axes_transform_representation_tolerance')
+    }
+
+    @classmethod
+    def _get_special_matched_key(cls, klass, vtk_attr_name):
```

Comment:
I generally don't mind docstrings for methods that are self-descriptive, such as the one above, but this one probably needs explanation.